src/main/java/net/minecraftforge/client/ItemDecoratorHandler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mojang.blaze3d.systems.RenderSystem;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import net.minecraft.client.gui.Font;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.client.event.RegisterItemDecorationsEvent;
+import net.minecraftforge.fml.ModLoader;
+import net.minecraftforge.fml.ModLoadingContext;
+import org.jetbrains.annotations.ApiStatus;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ApiStatus.Internal
+public final class ItemDecoratorHandler
+{
+    private final List<IItemDecorator> itemDecorators;
+
+    private static Map<Item, ItemDecoratorHandler> DECORATOR_LOOKUP = new Object2ObjectOpenHashMap<>();
+
+    private static final ItemDecoratorHandler EMPTY = new ItemDecoratorHandler();
+    private ItemDecoratorHandler()
+    {
+        this.itemDecorators = List.of();
```

Comment:
Use `ImmutableList.of()` here, this way it's an `ImmutableList` in both cases and the JVM does not need to deal with two different iterator implementations in `render`.